### ISSUE TITLE
Initial source state is null at startup which throws null pointer in listener implementation.

### ISSUE THREAD

**Issue by mrtinkz** [spring-projects/spring-statemachine#774]
Initial source state is null at startup which throws null pointer in listener implementation. The below was captured to show that the source is null before the statemachine moved to the initial state. This caused a null pointer in the registered listener on transitionStarted method. Although I could check for null and avoid this error, I doubt if that's the right way to do. Should a default state be assigned for the initial transitioning?

    `public void transitionStarted(Transition<NodeStates, NodeEvents> transition) {
		LOG.info(" transition started from {}  to {} due to {} ", transition.getSource().getId().name(),
				transition.getTarget().getId().name(), transition.getTrigger().getEvent().name());

	}`

`AbstractTransition[source = null, target = ObjectState[getIds() = [INIT], getClass() = class org.springframework.statemachine.state.ObjectState, hashCode() = 1453546898, toString() = AbstractState[id = INIT, pseudoState = org.springframework.statemachine.state.DefaultPseudoState @331a0e74, deferred = [], entryActions = [], exitActions = [], stateActions = [com.company.ua.consumer.config.StateMachineConfig$$Lambda$519 / 1344148545 @6ee700d9], regions = [], submachine = null]], kind = INITIAL, guard = null]`

**Comment by jvalkeal** [comment]
I know this is a bit annoying especially when logging these things like that. But having said that, when starting a machine and we're on an initial transition there is no state where it's coming from, so source is naturally null. Not really sure if there's any decent way to make this behaviour better but I'm up for suggestions?

**Comment by mrtinkz** [comment]
I agree, how about a sudo state like launch/onset/genesis? if none of this fit well I guess we can still live with a null or an Empty state which does not throw up null pointer.
